pinot-query-planner/src/main/java/org/apache/pinot/query/routing/WorkerManager.java

```diff
@@ -93,26 +94,31 @@ public void assignWorkerToStage(int stageId, StageMetadata stageMetadata, long r
               "Entry for server {} and table type: {} already exist!", serverEntry.getKey(), tableType);
         }
       }
-      stageMetadata.setServerInstances(new ArrayList<>(serverInstanceToSegmentsMap.keySet()));
+      stageMetadata.setServerInstances(new ArrayList<>(
+          serverInstanceToSegmentsMap.keySet()
+              .stream()
+              .map(server -> new VirtualServer(server, 0)) // for now, only use single virtual server
+              .collect(Collectors.toList())));
```

Comment:
this is only for the leaf stage correct? if so can we be more clear on the comments? thx

Reply:
👍 yes that's correct, the comment is a bit misleading